src/app/data-planes/sources.ts

```diff
@@ -23,12 +23,126 @@ export type DataplaneRulesCollectionSource = DataSourceResponse<DataplaneRulesCo
 const includes = <T extends readonly string[]>(arr: T, item: string): item is T[number] => {
   return arr.includes(item as T[number])
 }
+interface MetricRecord {
+  [key: string]: MetricRecord | string | number
+}
+
+// just does the inital response to JSON parsing
+const parse = (lines: string) => {
+  // for each line
+  return lines.trim().split('\n').filter((item) => {
+    // only use data prefixed with either cluster. http. or tcp.
+    return ['cluster.', 'http.', 'tcp.'].some(prop => item.startsWith(prop))
+  }).reduce((prev, item) => {
+    // split the `key: values` on the `:` and normalize the value
+    const [key, ...value] = item.trim().split(':')
+    const val = ((val) => {
+      try {
+        return JSON.parse(val)
+      } catch (e) {
+        return val
+      }
+    })(value.join(':').trim())
+
+    // walk the path creating `json.objects: value`
+    key.split('.').reduce((prev, item, i, arr) => {
+      if (i === arr.length - 1) {
+        // if this is the last segment in the path
+        // then just set the value
+        prev[item] = val
+        return prev
+      }
+      // otherwise add a {} if its not there already
+      // and return that as prev so we can continue appending
+      if (typeof prev[item] === 'undefined') {
+        const val = {}
+        prev[item] = val
+        return val
+      } else {
+        return prev[item]
+      }
+    }, prev)
+
+    return prev
+  }, {} as MetricRecord)
+}
+
+// lets you specify the things you are interested in
+const getTraffic = (json: ReturnType<typeof parse>, filter: (key: string) => boolean, protocols = ['http', 'tcp']) => {
+  return protocols.map((item) => {
+    return Object.entries(json[item]).filter(([key, _value]) => {
+      return filter(key)
+    }).map(([key, value]) => {
+      const parts = key.split('_')
+      // check see if the last thing is a number,
+      // if so its probably a port
+      const last = parts.pop()
+
+      let port
+      if (last && isNaN(parseInt(last))) {
+        // parts.push(last)
+      } else {
+        port = last
+      }
+      parts.push(last)
+      return {
+        metrics: value,
+        protocol: item,
+        name: parts.join('_'),
+        ...(port ? { port } : {}),
+      }
+    })
+  }).reduce((prev, item) => prev.concat(item))
+}
 
 export const sources = (api: KumaApi) => {
   return defineSources({
     '/meshes/:mesh/dataplanes/:name': async (params) => {
       return api.getDataplaneFromMesh(params)
     },
+    '/meshes/:mesh/dataplanes/:name/traffic': async (params) => {
+      const { mesh, name } = params
+
+      const res = await api.getDataplaneData({
+        mesh,
+        dppName: name,
+        dataPath: 'stats',
+      })
+
+      // parse the stuff
+      const json = parse(res)
```

Comment:
Parse the stuff from lines of `.` delimited strings, to a JSON structure using the above util